### ISSUE TITLE
Error on waiting for elastic search connection on Mac

### ISSUE THREAD

**Issue by danathala** [markshust/docker-magento#442]
**Description**
95 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHP CodeSniffer Config installed_paths set to ../../magento/magento-coding-standard,../../phpcompatibility/php-compatibility
dhanasekarp@Dhanasekars-Air mage2 % echo "127.0.0.1 ::1 magento.local" | sudo tee -a /etc/hosts
Password:
127.0.0.1 ::1 magento.local
dhanasekarp@Dhanasekars-Air mage2 % bin/setup magento.local
Stopping mage2_app_1           ... done
Stopping mage2_phpfpm_1        ... done
Stopping mage2_db_1            ... done
Stopping mage2_rabbitmq_1      ... done
Stopping mage2_redis_1         ... done
Stopping mage2_elasticsearch_1 ... done
Starting mage2_db_1            ... done
Starting mage2_redis_1         ... done
Starting mage2_rabbitmq_1      ... done
Starting mage2_elasticsearch_1 ... done
Starting mage2_phpfpm_1        ... done
Starting mage2_app_1           ... done
Waiting for connection to Elasticsearch...
......ERROR: 124
dhanasekarp@Dhanasekars-Air mage2 % 

**Steps To Reproduce**
1. 

**Expected Result**

**Actual Result**
Waiting for connection to Elasticsearch...
......ERROR: 124

**Comment by danathala** [comment]
issue fixed after increasing the timeout value. on bin/setup 
around Line 28
echo "Waiting for connection to Elasticsearch..."
bin/clinotty timeout 30 bash -c " - increased the value from 30- 100

**Comment by markshust** [comment]
Thanks @danathala, I re-opened this issue to take a closer look. We may need to up the default value here to prevent errors from occuring.

**Comment by markshust** [comment]
Noticed a few errors reported from certain users, where it took a while for Elasticsearch to start. I wonder if there is a way to globally change this, perhaps in a docker-compose file? Since I assume this could also happen during regular container starts.

**Comment by markshust** [comment]
I think this has been resolved some time ago, so closing this out. It's also possible your machine had less than 6GB RAM allocated in Docker Desktop.

**Comment by markshust** [comment]
Still occurring in most recent version, may need some updates rather than depending upon `sleep`:

```
Waiting for connection to Elasticsearch...
...
Waiting for Elasticsearch container to get initialized...
Could not connect to the Amqp Server.

In InstallCommand.php line 274:
                               
  Parameter validation failed  
                               

setup:install [--backend-frontname BACKEND-FRONTNAME] [--enable-debug-logging ENABLE-DEBUG-LOGGING] [--enable-syslog-logging ENABLE-SYSLOG-LOGGING] [--remote-storage-driver REMOTE-STORAGE-DRIVER] [--remote-storage-prefix REMOTE-STORAGE-PREFIX] [--remote-storage-bucket REMOTE-STORAGE-BUCKET] [--remote-storage-region REMOTE-STORAGE-REGION] [--remote-storage-key REMOTE-STORAGE-KEY] [--remote-storage-secret REMOTE-STORAGE-SECRET] [--consumers-wait-for-messages CONSUMERS-WAIT-FOR-MESSAGES] [--amqp-host AMQP-HOST] [--amqp-port AMQP-PORT] [--amqp-user AMQP-USER] [--amqp-password AMQP-PASSWORD] [--amqp-virtualhost AMQP-VIRTUALHOST] [--amqp-ssl AMQP-SSL] [--amqp-ssl-options AMQP-SSL-OPTIONS] [--key KEY] [--db-host DB-HOST] [--db-name DB-NAME] [--db-user DB-USER] [--db...
```

This is on an M1 Max with tons of RAM, so it's not an issue with performance on the host.

**Comment by markshust** [comment]
There is a workaround, as the above is not easily replicable. Stop all contains, remove all volumes & data, remove project folder and attempt re-install. 90%+ of the second time, everything installs properly.

**Comment by markshust** [comment]
I may have found something out related to this. Running `bin/setup-composer-auth` returns an error that says:

```
cp: /var/www/.composer/auth.json: No such file or directory
```

...even though that file exists, and is very clearly mounted as a host volume in the `docker-compose.yml` file.

**Comment by markshust** [comment]
I may have found something out related to this. Running `bin/setup-composer-auth` returns an error that says:

```
cp: /var/www/.composer/auth.json: No such file or directory
```

...even though that file exists, and is very clearly mounted as a host volume in the `docker-compose.yml` file.

![Screen Shot 2021-11-06 at 11 05 36 AM](https://user-images.githubusercontent.com/437029/140614337-00802413-f6a7-476c-9bd5-db73ad81a346.png)

**Comment by markshust** [comment]
Two updates on this:

1. `Could not connect to the Amqp Server.` is not an Elasticsearch error 😅. It is caused by RabbitMQ.
2. Magento could not connect to the RabbitMQ host (`rabbitmq`), at least on M1 Macs, due to this bug https://github.com/markshust/docker-magento/issues/566

The delay happening in #566 was causing the setup to error out, because there is a bug in Docker for Mac related to M1 chips that delays DNS responses for 5 seconds https://github.com/docker/for-mac/issues/5626

I'll have fixes for both of these issues very soon, now that I was finally able to find fixes for these seemingly rando bugs.

**Comment by markshust** [comment]
Fixed in `41.0.0`

**Comment by markshust** [comment]
This is still happening 😅

So, the network issue is resolved, but I found out the problem is that RabbitMQ doesn't have enough time to boot up before the setup starts.

You can replicate this error by running:

```
bin/stop
bin/remove
bin/setup magento.test
```

You can also verify how long it takes for the RabbitMQ to start by running:

```
docker-compose run --rm -it rabbitm
```

It takes a few moments to start, so I'll add the sleep back in until we can find another way to do this.

**Comment by markshust** [comment]
Properly fixed (for now with `sleep`) in `41.0.1`.

**Comment by Asitis** [comment]
@markshust I'm having all kinds of issues with my setup, so I decided to remove everything and start over with a clean slate on Ubuntu. The `bin/setup` command stops after that timeout on `Waiting for connection to Elasticsearch`, the cli output just stops and that's that.
I see in the `bin/setup` that the timeout is already set to 100 as per danathala's suggestion, and I see the `sleep` you added in the latest version, but, it still doesn't work.
(And yes, I did do all the extra Linux setup to resolve the hostname etc). Do you have any tips how to handle or debug this further?
